R/add_lcsi.R

```diff
@@ -1,20 +1,179 @@
-#                                                      TRUE ~ NA_character_ ))
-#
-# }
+#' Add LCSI
+#'
+#' Function to calculate Livelihood Coping Strategy Index (LCSI)
+#'
+#' @param .dataset A dataframe with the ten LCSI variables needed for analysis.
+#' @param lcsi_stress_vars A vector of character values that are the column names for the four stress LCSI variables.
+#' @param lcsi_crisis_vars A vector of character values that are the column names for the three crisis LCSI variables.
+#' @param lcsi_emergency_vars A vector of character values that are the column names for the thre emergency LCSI variables.
+#' @param yes_val A character value in the dataset associated with "Yes, used this coping strategy in the last 30 days."
+#' @param no_val A character value in the dataset associated with "No, have not used this coping strategy in the last 30 days."
+#' @param exhausted_val A character value in the dataset associated with "No, haven't used in the last 30 days because I've exhausted this coping strategy in the last 6 or 12 months."
+#' @param not_applicable_val A character value in the dataset associated with "This coping strategy is not applicable for the household.
+#'
+#' @return Returns a dataframe with added columns for LCSI indicators.
+#' @export
+#'
+#' @examples{
+#'   input_data1 <- data.frame(
+#' stress1 =             c("No", "No", "Exhausted", "Not Applicable", "No"),
+#' stress2 =             c("No", "Yes", "Not Applicable", "No", "No"),
+#' stress3 =             c("Not Applicable", "Not Applicable", "Yes", "No", "No"),
+#' stress4 =             c("Not Applicable", "No", "Yes", "Yes", "No"),
+#' crisis1 =             c("No", "Not Applicable", "Yes", "Exhausted", "No"),
+#' crisis2 =             c("No", "No", "No", "No", "No"),
+#' crisis3 =             c("No", "No", "Yes", "Not Applicable", "No"),
+#' emergency1 =          c("No", "Not Applicable", "Not Applicable", "No", "No"),
+#' emergency2 =          c("No", "Not Applicable", "Yes", "Not Applicable", "No"),
+#' emergency3 =          c("Not Applicable", "No", "Not Applicable", "No", "Exhausted"))
+#'
+#' add_lcsi(.dataset = input_data1,
+#' lcsi_stress_vars = c("stress1", "stress2", "stress3", "stress4"),
+#' lcsi_crisis_vars = c("crisis1", "crisis2", "crisis3"),
+#' lcsi_emergency_vars = c("emergency1", "emergency2", "emergency3"),
+#' yes_val = "Yes",
+#' no_val = "No",
+#' exhausted_val = "Exhausted",
+#' not_applicable_val = "Not Applicable")
+#'
+#' }
+add_lcsi <- function(.dataset,
+                     lcsi_stress_vars,
+                     lcsi_crisis_vars,
+                     lcsi_emergency_vars,
+                     yes_val = NULL,
+                     no_val = NULL,
+                     exhausted_val = NULL,
+                     not_applicable_val = NULL
+) {
+
+df <- .dataset
+
+  #If NULL, set a standard default values for LCSI responses.
+
+  if(is.null(yes_val)) {yes_val = "1"}
+  if(is.null(yes_val)) {no_val = "2"}
```

Comment:
I think this should be:
if(is.null(no_val)) {no_val = "2"}
if(is.null(exhausted_val)) {exhausted_val = "3"}
if(is.null(not_applicable_val)) {not_applicable_val = "4"}
[You used yes_val for all 4 checks]